src/Arcus.EventGrid.Publishing/EventGridPublisherBuilder.cs

```diff
@@ -56,22 +59,77 @@ private EventGridPublisherBuilder(Uri topicEndpoint)
+        /// </summary>
+        /// <param name="topicEndpoint">The URL of the custom Event Grid topic.</param>
+        /// <param name="logger">The logger instance to write dependency telemetry during the interaction with the Azure Event Grid resource.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="topicEndpoint"/> is blank.</exception>
+        /// <exception cref="UriFormatException">Thrown when the <paramref name="topicEndpoint"/> is not a correct URL.</exception>
+        /// <example>
+        /// Shows how a <see cref="EventGridPublisher"/> should be created.
+        /// <code>
+        /// EventGridPublisher myPublisher =
+        ///     EventGridPublisherBuilder
+        ///         .ForTopic("http://myTopic", logger)
+        ///         .UsingAuthenticationKey("myAuthenticationKey")
+        ///         .Build();
+        /// </code>
+        /// </example>
+        public static EventGridPublisherBuilder ForTopic(string topicEndpoint, ILogger logger)
+        {
+            Guard.NotNullOrWhitespace(topicEndpoint, nameof(topicEndpoint), "Requires a non-blank topic endpoint for the Azure Event Grid publisher");
+            Guard.For(() => Uri.IsWellFormedUriString(topicEndpoint, UriKind.Absolute) == false, 
+                new ArgumentException("Requires a URI-valid topic endpoint for the Azure Event Grid publisher", nameof(topicEndpoint)));
+
+            return ForTopic(topicEndpoint, logger, configureOptions: null);
+        }
+        
+        /// <summary>
+        /// Specifies the custom Event Grid <paramref name="topicEndpoint"/> for which a <see cref="IEventGridPublisher"/> will be created.
+        /// </summary>
+        /// <param name="topicEndpoint">The URL of the custom Event Grid topic.</param>
+        /// <param name="logger">The logger instance to write dependency telemetry during the interaction with the Azure Event Grid resource.</param>
+        /// <param name="configureOptions">The additional function to configure optional settings on the <see cref="IEventGridPublisher"/>.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="topicEndpoint"/> is blank.</exception>
+        /// <exception cref="UriFormatException">Thrown when the <paramref name="topicEndpoint"/> is not a correct URL.</exception>
+        /// <example>
+        /// Shows how a <see cref="EventGridPublisher"/> should be created.
+        /// <code>
+        /// EventGridPublisher myPublisher =
+        ///     EventGridPublisherBuilder
+        ///         .ForTopic("http://myTopic", logger)
+        ///         .UsingAuthenticationKey("myAuthenticationKey")
+        ///         .Build();
+        /// </code>
+        /// </example>
+        public static EventGridPublisherBuilder ForTopic(string topicEndpoint, ILogger logger, Action<EventGridPublisherOptions> configureOptions)
+        {
```

Comment:
Shouldn't we guard for a null action or null logger?

Reply:
Well, the logging is optional and so is the configurable options function, so I would say no.